frontend/app/model/RichEvent.scala

```diff
@@ -148,17 +143,23 @@ object RichEvent {
   }
 
   case class MasterclassEvent(event: EBEvent, data: Option[MasterclassData]) extends RichEvent {
-    val imgUrl = data.flatMap(_.images.headOption).flatMap(_.file)
-      .getOrElse(fallbackImage)
-      .replace("http://static", "https://static-secure")
-
-    val imageMetadata = None
 
-    val availableWidths = ""
+    val imgOpt = data.flatMap(_.images.headOption).flatMap { asset =>
+      asset.file.map { imgUrl =>
+        ResponsiveImageGroup(
+          altText = event.name.toString(),
+          availableImages = Seq(ResponsiveImage(
+            path = imgUrl.replace("http://static", "https://static-secure"),
+            width = asset.typeData.get("width").map(_.toInt).getOrElse(460)
+          ))
+        )
+      }
+    }
 
-    val srcsetOpt = None
+    val imageMetadata = None
 
-    val socialImgUrl = imgUrl
+    //TODO: Should be Option[String]
+    val socialImgUrl = imgOpt.map(_.defaultImage).getOrElse("")
```

Comment:
This is a bit naff, returning an empty string. But switching this to an `Option` is more refactoring than feels appropriate in this PR and returning an empty string for the `socialImageUrl` is OK in the short term.